frontend/src/js/actions/workspaces/takeOwnershipOfWorkspace.ts

```diff
@@ -0,0 +1,33 @@
+import { updateWorkspaceOwner } from '../../services/WorkspaceApi';
+import { getWorkspacesMetadata } from './getWorkspacesMetadata';
+import { ThunkAction } from 'redux-thunk';
+import { AppAction, AppActionType, WorkspacesAction } from '../../types/redux/GiantActions';
+import { GiantState } from '../../types/redux/GiantState';
+import { getWorkspace } from './getWorkspace';
+
+export function takeOwnershipOfWorkspace(
+    id: string,
+    user: string,
+): ThunkAction<void, GiantState, null, WorkspacesAction | AppAction> {
+    return dispatch => {
+        return updateWorkspaceOwner(id, user)
+            .then(res => {
+                dispatch(getWorkspacesMetadata());
+                // We need to fire off this second call because "is public" metadata as
+                // displayed in the "current workspace" view is accessed from the currentWorkspace state,
+                // so we need to refresh it. It's also in the workspacesMetadata state, but we don't want to
+                // search through that array to find the current workspace: currentWorkspace should always
+                // be the canonical source of info for that.
```

Comment:
is this comment accurate for 'take ownership' - if so maybe update it so it's not talking about "is public"